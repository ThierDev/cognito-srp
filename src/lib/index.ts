export * from './ClientPasswordChallenge';
export * from './ServerPasswordChallenge';
export * from './Session';
export * from './UserPool';
