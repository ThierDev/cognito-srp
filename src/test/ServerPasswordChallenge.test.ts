import { ServerPasswordChallenge } from '../lib';
import { secret, poolname, serverUser, publicKey } from './constants';

describe('ServerPasswordChallenge', () => {
  let challenge: ServerPasswordChallenge;

  beforeAll(() => {
    challenge = new ServerPasswordChallenge(poolname, serverUser, secret);
  });

  it('should calculate A', () => {
    const B = challenge.calculateB();
    expect(B).toMatchSnapshot();
  });

  it('should get a session', () => {
    const session = challenge.getSession(publicKey);
    expect(session.getHkdf()).toMatchSnapshot();
  });
});
