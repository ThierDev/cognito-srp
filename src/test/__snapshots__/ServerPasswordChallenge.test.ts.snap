// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServerPasswordChallenge should calculate A 1`] = `
Object {
  "data": Array [
    162,
    104,
    73,
    169,
    189,
    17,
    201,
    103,
    187,
    19,
    87,
    64,
    230,
    244,
    25,
    255,
    138,
    20,
    12,
    117,
    5,
    129,
    172,
    162,
    100,
    135,
    238,
    219,
    145,
    227,
    169,
    185,
    6,
    64,
    111,
    147,
    71,
    212,
    26,
    62,
    101,
    68,
    160,
    219,
    139,
    13,
    168,
    201,
    204,
    208,
    67,
    253,
    174,
    47,
    50,
    112,
    101,
    51,
    183,
    220,
    43,
    104,
    9,
    82,
    3,
    45,
    248,
    26,
    15,
    176,
    49,
    71,
    205,
    38,
    39,
    86,
    59,
    56,
    235,
    132,
    193,
    46,
    103,
    50,
    131,
    140,
    168,
    81,
    80,
    34,
    135,
    247,
    136,
    158,
    195,
    125,
    27,
    127,
    231,
    236,
    234,
    199,
    154,
    5,
    136,
    196,
    75,
    237,
    210,
    185,
    150,
    114,
    177,
    222,
    226,
    229,
    237,
    250,
    70,
    122,
    36,
    218,
    248,
    9,
    76,
    10,
    248,
    86,
    177,
    41,
    100,
    149,
    125,
    91,
    217,
    100,
    132,
    82,
    69,
    222,
    173,
    156,
    186,
    56,
    134,
    216,
    31,
    252,
    14,
    74,
    39,
    16,
    194,
    127,
    240,
    94,
    198,
    44,
    91,
    145,
    29,
    53,
    177,
    242,
    9,
    54,
    33,
    250,
    102,
    148,
    83,
    49,
    157,
    186,
    11,
    55,
    202,
    178,
    211,
    43,
    11,
    233,
    31,
    253,
    130,
    208,
    27,
    118,
    6,
    253,
    100,
    216,
    63,
    238,
    39,
    201,
    56,
    207,
    172,
    30,
    2,
    222,
    24,
    246,
    78,
    5,
    104,
    117,
    133,
    110,
    13,
    137,
    92,
    213,
    117,
    77,
    171,
    1,
    39,
    194,
    183,
    113,
    8,
    4,
    115,
    208,
    90,
    205,
    149,
    2,
    88,
    123,
    196,
    230,
    211,
    139,
    227,
    191,
    106,
    119,
    122,
    176,
    158,
    165,
    232,
    42,
    109,
    2,
    2,
    92,
    40,
    134,
    70,
    1,
    96,
    52,
    181,
    234,
    231,
    22,
    203,
    14,
    99,
    213,
    204,
    195,
    8,
    139,
    32,
    128,
    102,
    91,
    177,
    8,
    230,
    56,
    205,
    187,
    195,
    60,
    193,
    136,
    72,
    171,
    194,
    33,
    166,
    206,
    103,
    16,
    75,
    132,
    146,
    190,
    123,
    62,
    135,
    63,
    254,
    25,
    84,
    198,
    252,
    245,
    184,
    97,
    105,
    164,
    109,
    235,
    190,
    91,
    219,
    66,
    141,
    59,
    144,
    127,
    38,
    29,
    28,
    17,
    159,
    41,
    200,
    118,
    184,
    16,
    166,
    75,
    125,
    20,
    247,
    220,
    220,
    132,
    239,
    157,
    61,
    235,
    6,
    91,
    100,
    186,
    245,
    52,
    22,
    103,
    32,
    145,
    217,
    128,
    19,
    149,
    34,
    105,
    176,
    62,
    63,
    154,
    63,
    206,
    57,
    60,
    214,
    207,
    19,
    65,
    121,
    99,
    224,
    128,
    150,
    116,
    99,
    136,
    246,
    12,
    213,
    192,
    178,
    104,
    103,
    4,
  ],
  "type": "Buffer",
}
`;

exports[`ServerPasswordChallenge should get a session 1`] = `"79ffcab44257a27e2a8a9874d4b35e3e"`;
